packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -619,8 +622,13 @@ class PageView extends StatefulWidget {
   final bool pageSnapping;
 
   /// Called whenever the page in the center of the viewport changes.
+  ///
+  /// It is called before [pageDidChanged], in the middle of scroll animation.
```

Comment:
Isn't it called at the start of the transition?

Reply:
Actually it is called in the middle.According to the source code in `page_view.dart` line685-691
```
        if (notification.depth == 0 && widget.onPageChanged != null && notification is ScrollUpdateNotification) {
          final PageMetrics metrics = notification.metrics;
          final int currentPage = metrics.page.round(); 
          if (currentPage != _lastOnPageChangedIndex) {
            _lastOnPageChangedIndex = currentPage;
            widget.onPageChanged(currentPage);
          }
```
Only if the [metrics.page] is beyond 0.5, you can detect the page changing;
And this point is proved in the test script - `page_view_test.dart`
BTW, sorry for the grammatical mistakes.

Reply:
You're right! Thanks for checking.